src/drivers/sdl/lv_sdl_window.c

```diff
@@ -364,16 +364,21 @@ static void texture_resize(lv_display_t * disp)
     }
     if(dsc->texture) SDL_DestroyTexture(dsc->texture);
 
-#if LV_COLOR_DEPTH == 32
-    SDL_PixelFormatEnum px_format =
-        SDL_PIXELFORMAT_RGB888; /*same as SDL_PIXELFORMAT_RGB888, but it's not supported in older versions*/
-#elif LV_COLOR_DEPTH == 24
-    SDL_PixelFormatEnum px_format = SDL_PIXELFORMAT_BGR24;
-#elif LV_COLOR_DEPTH == 16
-    SDL_PixelFormatEnum px_format = SDL_PIXELFORMAT_RGB565;
-#else
-#error("Unsupported color format")
-#endif
+    SDL_PixelFormatEnum px_format;
+
+    switch(lv_color_format_get_size(lv_display_get_color_format(disp)) * 8) {
+        case 32:
```

Comment:
Once we are updating it please use an ARGB format here if LVGL is configured to ARGB8888.

Reply:
I am not sure what exactly you want me to do. 

The only thing that has changed is the removal of the macros in favor of using the color format. Basically it was testing for 32 bit, 24 bit and 16 bit color using macros that really don't need to be set anymore and could actually differ than what the real color depth is. So, I switched it over to using the color format instead. I am testing the color formats to check how many bits there are and setting the same exact SDL color formats that were using before.